hbase-http/src/test/java/org/apache/hadoop/hbase/http/TestSpnegoHttpServer.java

```diff
@@ -234,4 +239,15 @@ public void testMissingConfigurationThrowsException() throws Exception {
     customServer.addJerseyResourcePackage(JerseyResource.class.getPackage().getName(), "/jersey/*");
     customServer.start();
   }
+
+  @Test
+  public void testEndpointWhitelist() throws IOException {
+    URL url = new URL(getServerURL(server), "/jmx");
+    HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+    assertEquals(HttpURLConnection.HTTP_UNAUTHORIZED, conn.getResponseCode());
+//    FIXME: hadoop >= 3.4.0 (or hadoop with HADOOP-18666) is needed to test with below lines.
+//    url = new URL(getServerURL(server), "/prometheus");
+//    conn = (HttpURLConnection) url.openConnection();
+//    assertEquals(HttpURLConnection.HTTP_OK, conn.getResponseCode());
```

Comment:
I added a test case but it cannot be tested properly if you don't use hadoop with HADOOP-18666 :cry: .